spec/jobs/fix_primary_emails_for_staged_users_spec.rb

```diff
@@ -1,25 +1,35 @@
 require 'rails_helper'
 
 RSpec.describe Jobs::FixPrimaryEmailsForStagedUsers do
-  it 'should clean up duplicated staged users' do
-    common_email = 'test@reply'
-
-    staged_user = Fabricate(:user, staged: true, active: false)
-    staged_user2 = Fabricate(:user, staged: true, active: false)
-    staged_user3 = Fabricate(:user, staged: true, active: false)
+  let(:common_email)  { 'test@reply' }
+  let(:staged_user)   { Fabricate(:user, staged: true, active: false) }
+  let(:staged_user2)  { Fabricate(:user, staged: true, active: false) }
+  let(:staged_user3)  { Fabricate(:user, staged: true, active: false) }
+  let(:active_user)   { Fabricate(:coding_horror) }
 
+  before do
     [staged_user, staged_user2, staged_user3].each do |user|
       user.email_tokens = [Fabricate.create(:email_token, email: common_email, user: user)]
     end
 
-    active_user = Fabricate(:coding_horror)
-
     UserEmail.delete_all
+  end
 
+  it 'should clean up duplicated staged users' do
     expect { described_class.new.execute_onceoff({}) }
       .to change { User.count }.by(-2)
 
     expect(User.all).to contain_exactly(Discourse.system_user, staged_user, active_user)
     expect(staged_user.reload.email).to eq(common_email)
   end
+
+  it 'should keep the duplicated staged user with posts' do
```

Comment:
Ah OK I see what is happening. So both the original and duplicated staged users have posted? If that is the case, we're need to transfer all the posts to the original staged user and nuke the duplicated user.